File: artifacts/mimalloc/decompose/src/mi_memid_t.c
typedef struct mi_memid_s
{
  union 
  {
    mi_memid_os_info_t os;
    mi_memid_arena_info_t arena;
  } mem;
  bool is_pinned;
  bool initially_committed;
  bool initially_zero;
  mi_memkind_t memkind;
} mi_memid_t
/*** DEPENDENCIES:
typedef struct mi_memid_arena_info
{
  size_t block_index;
  mi_arena_id_t id;
  bool is_exclusive;
} mi_memid_arena_info_t
----------------------------
typedef struct mi_memid_os_info
{
  void *base;
  size_t size;
} mi_memid_os_info_t
----------------------------
typedef enum mi_memkind_e
{
  MI_MEM_NONE,
  MI_MEM_EXTERNAL,
  MI_MEM_STATIC,
  MI_MEM_OS,
  MI_MEM_OS_HUGE,
  MI_MEM_OS_REMAP,
  MI_MEM_ARENA
} mi_memkind_t
----------------------------
***/

File: artifacts/mimalloc/manual_debug/src/mi_memid_t.rs
use crate::*;
use std::mem;


#[derive(Clone)]
pub struct MiMemidS {
    pub mem: MiMemidUnion,
    pub is_pinned: bool,
    pub initially_committed: bool,
    pub initially_zero: bool,
    pub memkind: MiMemkindT,
}

#[derive(Clone)]
pub enum MiMemidUnion {
    Os(MiMemidOsInfo),
    Arena(MiMemidArenaInfo),
}

